docs/transforms.md

```diff
@@ -0,0 +1,114 @@
+## Transforms
+
+Transforms are operations applied on single records (i.e. multi-message transforms are not yet supported). 
+
+They are optional parts of a pipeline and, if present, they come in the form of processors (more information about 
+processor can be found in the [Architecture documentation](/docs/architecture.md)). As processors, transforms can be 
+attached either to a connector or to a pipeline. In other words, transforms can transform:
+1. Records coming out of a source connector (if the transform is attached to a source connector)
+2. Records coming into a destination connector (if the transform is attached to a destination connector)
+3. All records flowing through a pipeline (if the transform is attached to a pipeline)
+
+We'll go a little deeper into the case of a pipeline transform. At a high-level, source connectors, pipeline transforms
+and destination connector are laid out as in this image:
+
+![Pipeline](data/pipeline.svg)
```

Comment:
I see we are simplifying here and only include pipeline transforms in the drawing, I guess it's for the sake of simplicity. I'm not sure this is the right place to simplify though, I think people will come here to dive into transforms and get the full information about them. I also think the drawing can be misleading since we first explain the difference between transforms attached to a source, destination and pipeline, while the picture only shows pipeline transforms; as a user, I'd be left with questions how the other two fit in.

The very least I expect here is that we point the user to the [pipeline structure](https://github.com/ConduitIO/conduit/blob/main/docs/pipeline_semantics.md#pipeline-structure) doc where the structure is described in detail. But _ideally_ I think we should include source/destination transforms to show the full picture.

Reply:
I won't block because of this, but I honestly don't see a reason why we would not show the whole picture in a doc specifically talking about transforms 🤷 For you it might have been obvious where source and destination transforms fit in and had problems understanding pipeline transforms, for someone else it might be the other way around.

Reply:
If you feel very strongly about that, I can definitely change. As for me, I don't think that any option is much ahead of the other one.: )